### PR TITLE
Add deprecated flag

### PR DESCRIPTION
**Pull request by dgomezc**
# PR checklist

**Quick summary of changes**
  - Add deprecated message in framework cards.
  - Add deprecated message in framework detail pages.
 
**Which issue does this PR relate to?**
#3928 - Need official designs for deprecation flags

**Applies to the following platforms:**

| UWP              | WPF              | WinUI           |
| :--------------- | :--------------- | :---------------|
| Yes | Yes | No |

Sscreenshots:
Framework selection page
Frawemork detail page

### CODE REVIEW

**Comment by mrlacey** [comment]
When I added the original support for a deprecated flag, it was done with the flexibility that it could be used for any text or future "flag" that may be needed".
This change appears to have hard-coded the warning icon but this may not be appropriate for all future "flags".
Was making the icon configurable considered?

I'm ok if the answer is "we decided against this but could add it in the future if needed."
What I'm interested in is the decision-making process and the options that were considered. There is no documentation of the process or the reasoning behind any decisions. I have ideas for future functionality that will require the use of different "flags" and now icons. If work has been done that will make this easier or harder or investigations have discovered things (such as edge-cases) that need to be considered as part of an implementation it will save time in the future (regardless of who implements them or why) if these are documented now.

**Comment by dgomezc** [comment]
We only requested feedback from the design team for the deprecated flag. We are not sure if the same design is valid for other flags. 

We used "Flag" property because is was already created but we think this should use a dedicated "IsDeprecated" tag. If we add other flags we should ask the design team for the correct way to show them and adapt the code to include those flags.

File: code/src/UI/ViewModels/Common/DataItems/FrameworkMetaDataViewModel.cs [hunk 1]
```diff
@@ -33,12 +33,12 @@ public FrameworkMetaDataViewModel(MetadataInfo metadataInfo)
             Order = metadataInfo.Order;
             MetadataType = metadataInfo.MetadataType;
             Licenses = metadataInfo.LicenseTerms?.Select(l => new LicenseViewModel(l));
-            Flag = GetLocalizedFlag(metadataInfo.Tags.FirstOrDefault(t => t.Key.Equals("wts.flag")).Value?.ToString());
+            Deprecated = bool.TryParse(metadataInfo.Tags.FirstOrDefault(t => t.Key.Equals("deprecated")).Value?.ToString(), out bool isDeprecated);
         }
 
-        private string GetLocalizedFlag(string flagValue)
+        private string GetLocalizedTag(string flagValue)
```
Comment (mrlacey) [inline code/src/UI/ViewModels/Common/DataItems/FrameworkMetaDataViewModel.cs:39]: Where is this used?
What are the "tag"s that this relates to?

**Comment by mrlacey** [comment]
What's the reason for this latest change?
The flag capability was originally created to be flexible because I knew this would be reused for other things in the future.
I know (because you've said above) that designs are only approved for the deprecated flag but I don't see why the existing flag capability could not have been used for this.

With this change is looks like:
- You've done extra work for no immediate benefit.
- You've removed the capability for future flexibility, meaning that additional work will be required in the future when we need a flag for other things.


I'm assuming there's something I'm missing in the above decision-making. Please fill me in.

**Comment by sibille** [comment]
I actually liked the flexibility of the flags property, but the design proposal we got from the design team is very specific to flagging a deprecation, that's why we decided to create a specific tag for this.
We were thinking about leaving the flags property, but as we currently do not have any flags we decided to remove it (for now). What other flags are you thinking about? Preview, New?

**Review by sibille — APPROVED** [review]

**Pull request merged**